runtime/pool.go

```diff
@@ -0,0 +1,37 @@
+package runtime
+
+import "sync"
+
+type Pool struct {
```

Comment:
Please add a design annotation for this package of struct. Otherwise, other could hardly understand what you mean exactly.